src/jit.c

```diff
@@ -0,0 +1,1557 @@
+    LSP_LDPX = 0xa9400000U,  // 1010_1001_0100_0000_0000_0000_0000_0000
+    /* UnconditionalBranchOpcode */
+    BR_BR = 0xd61f0000U,   // 1101_0110_0001_1111_0000_0000_0000_0000
+    BR_BLR = 0xd63f0000U,  // 1101_0110_0011_1111_0000_0000_0000_0000
+    BR_RET = 0xd65f0000U,  // 1101_0110_0101_1111_0000_0000_0000_0000
+    /* UnconditionalBranchImmediateOpcode */
+    UBR_B = 0x14000000U,  // 0001_0100_0000_0000_0000_0000_0000_0000
+    /* ConditionalBranchImmediateOpcode */
+    BR_Bcond = 0x54000000U,
+    /* DP2Opcode */
+    DP2_UDIV = 0x1ac00800U,  // 0001_1010_1100_0000_0000_1000_0000_0000
+    DP2_LSLV = 0x1ac02000U,  // 0001_1010_1100_0000_0010_0000_0000_0000
+    DP2_LSRV = 0x1ac02400U,  // 0001_1010_1100_0000_0010_0100_0000_0000
+    DP2_ASRV = 0x1ac02800U,  // 0001_1010_1100_0000_0010_1000_0000_0000
+    /* DP3Opcode */
+    DP3_MADD = 0x1b000000U,  // 0001_1011_0000_0000_0000_0000_0000_0000
+    DP3_MSUB = 0x1b008000U,  // 0001_1011_0000_0000_1000_0000_0000_0000
+    /* MoveWideOpcode */
+    MW_MOVN = 0x12800000U,  // 0001_0010_1000_0000_0000_0000_0000_0000
+    MW_MOVZ = 0x52800000U,  // 0101_0010_1000_0000_0000_0000_0000_0000
+    MW_MOVK = 0x72800000U,  // 0111_0010_1000_0000_0000_0000_0000_0000
+};
+
+enum condition {
+    COND_EQ,
+    COND_NE,
+    COND_HS,
+    COND_LO,
+    COND_GE = 10,
+    COND_LT = 11,
+};
+#endif
+
+enum vm_reg {
+    VM_REG_0 = 0,
+    VM_REG_1,
+    VM_REG_2,
+    VM_REG_3,
+    VM_REG_4,
+    VM_REG_5,
+    VM_REG_6,
+    VM_REG_7,
+    VM_REG_8,
+    VM_REG_9,
+    VM_REG_10,
+    N_VM_REGS,
+};
+
+enum operand_size {
+    S8,
+    S16,
+    S32,
+};
+
+#if defined(__x86_64__)
+/* There are two common x86-64 calling conventions, discussed at:
+ * https://en.wikipedia.org/wiki/X64_calling_conventions#x86-64_calling_conventions
+ *
+ * Please note: R12 is an exception and is *not* being used. Consequently, it
+ * is omitted from the list of non-volatile registers for both platforms,
+ * despite being non-volatile.
+ */
+#if defined(_WIN32)
+static const int nonvolatile_reg[] = {RBP, RBX, RDI, RSI, R13, R14, R15};
+static const int parameter_reg[] = {RCX, RDX, R8, R9};
+#define RCX_ALT R10
+static const int register_map[] = {
+    RAX, R10, RDX, R8, R9, R14, R15, RDI, RSI, RBX, RBP,
+};
+#else
+#define RCX_ALT R9
+static const int nonvolatile_reg[] = {RBP, RBX, R13, R14, R15};
+static const int parameter_reg[] = {RDI, RSI, RDX, RCX, R8, R9};
+static const int temp_reg[] = {RAX, RBX, RCX};
+static const int register_map[] = {
+    RAX, RDI, RSI, RDX, R9, R8, RBX, R13, R14, R15, RBP,
+};
+#endif
+#elif defined(__aarch64__)
+/* callee_reg - this must be a multiple of two because of how we save the stack
+ * later on. */
+static const int callee_reg[] = {R19, R20, R21, R22, R23, R24, R25, R26};
+/* parameter_reg (Caller saved registers) */
+static const int parameter_reg[] = {R0, R1, R2, R3, R4};
+static const int temp_reg[] = {R6, R7, R8};
+/* Temp register for immediate generation */
```

Comment:
Use `enum` for both `temp_imm_reg` and `temp_div_reg` because `const` is just a qualifier and the variables still occupy space. That is,
```c
enum {
    temp_imm_reg = R24, /* Temp register for immediate generation */
    temp_div_reg = R25, /* Temp register for division results */
};
```